pilot/pkg/model/push_context.go

```diff
@@ -660,12 +660,16 @@ func (ps *PushContext) Services(proxy *Proxy) []*Service {
 	return out
 }
 
-// ServiceForHostname returns the service associated with a given hostname following SidecarScope
-func (ps *PushContext) ServiceForHostname(proxy *Proxy, hostname host.Name) *Service {
+// ServiceForHostnameAndNamespace returns the service associated with a given hostname in the given namespace following SidecarScope
+func (ps *PushContext) ServiceForHostnameAndNamespace(proxy *Proxy, hostname host.Name, namespace string) *Service {
 	if proxy != nil && proxy.SidecarScope != nil {
 		return proxy.SidecarScope.servicesByHostname[hostname]
 	}
 
+	if namespace != "" {
```

Comment:
I am not sure we need this. Probably the function should be moved to be on proxy.SidecarScope, because this code should never trigger as `proxy.SidecarScope.servicesByHostname` will always trigger first.

I think for your case you actually want a different function that is directly fetching from ` ps.ServiceIndex.HostnameAndNamespace` without the first call to `proxy.SidecarScope.servicesByHostname`.

Reply:
Otherwise, I suspect you may lookup `foo/google.com` but actually get `bar/google.com` if the sidecar scope imports bar's

Reply:
okay, then I can just use the `ps.ServiceIndex.HostnameAndNamespace` directly, no need for another function.